Pagination/MongoTrait/FilterTypeStrategy/Strategies/TranslatedFilterStrategy.php

```diff
@@ -39,16 +40,87 @@ public function support($type)
      */
     public function generateFilter($name, $value, $documentName)
     {
-        $metaData = $this->documentManager->getClassMetadata($documentName);
-        $collectionName = $metaData->getCollection();
+        $collection = $this->documentManager->getDocumentCollection($documentName);
+        $collectionName = $collection->getName();
         $dataBase = $this->documentManager->getDocumentDatabase($documentName);
-        $criteria = '{"'.$name.'.*.value": /.*'.$value.'.*/i}';
+        $criteria = array($name.'.*.value' =>  new \MongoRegex('/.*'.$value.'.*/i'));
+        $filter = $this->parseCriteria($criteria, $collectionName, $dataBase);
+        if (!empty($filter)) {
+
+            return $filter;
+        }
+
+        return null;
+    }
+
+
+    /**
+     * @param array    $criteria
+     * @param string   $collectionName
+     * @param Database $dataBase
+     *
+     * @return array
+     */
+    protected function parseCriteria($criteria, $collectionName, $dataBase)
+    {
+        foreach ($criteria as $key => $value) {
+            if(strpos($key, "*") !== false){
+                $res = $this->generateCriteria($key, $value, $collectionName, $dataBase);
+                if (!empty($res)) {
+                    $criteria['$or'] = $res;
```

Comment:
I think there is a problem here: the second iteration will override the result of the first, the third will override the second, and so on